### ISSUE TITLE
HDF5 Not Found in TravisCI, Doesn't Fail

### ISSUE THREAD

**Issue by stimpsonsg** [CASL/Futility#132]
So we've got a problem with the tests Futility is doing (see link and snippet below), but basically the second configure is failing because it can't find HDF5.  However, nothing indicates the failure on the merge request, so it seems like everything is fine, when it is, in fact, very much not.

https://travis-ci.org/CASL/Futility/builds/425505500?utm_source=github_status&utm_medium=notification

```
Processing enabled TPL: HDF5 (enabled explicitly, disable with -DTPL_ENABLE_HDF5=OFF)
-- HDF5_LIBRARY_NAMES='hdf5_hl;hdf5;hdf5_cpp;hdf5_fortran'
-- Searching for libs in HDF5_LIBRARY_DIRS='/tools/vera/gcc-5.4.0/tpls/opt/hdf5-1.8.10/lib'
-- Searching for a lib in the set "hdf5_hl":
--   Searching for lib 'hdf5_hl' ...
-- NOTE: Did not find a lib in the lib set "hdf5_hl" for the TPL 'HDF5'!
-- ERROR: Could not find the libraries for the TPL 'HDF5'!
-- TIP: If the TPL 'HDF5' is on your system then you can set:
     -DHDF5_LIBRARY_DIRS='<dir0>;<dir1>;...'
   to point to the directories where these libraries may be found.
   Or, just set:
     -DTPL_HDF5_LIBRARIES='<path-to-libs0>;<path-to-libs1>;...'
   to point to the full paths for the libraries which will
   bypass any search for libraries and these libraries will be used without
   question in the build.  (But this will result in a build-time error
   if not all of the necessary symbols are found.)
-- ERROR: Failed finding all of the parts of TPL 'HDF5' (see above), Aborting!
-- NOTE: The find module file for this failed TPL 'HDF5' is:
     /mnt/Futility/./cmake/tpl/FindTPLHDF5.cmake
   which is pointed to in the file:
     /mnt/Futility/./TPLsList.cmake
TIP: Even though the TPL 'HDF5' was explicitly enabled in input,
it can be disabled with:
  -DTPL_ENABLE_HDF5=OFF
which will disable it and will recursively disable all of the
downstream packages that have required dependencies on it.
When you reconfigure, just grep the cmake stdout for 'HDF5'
and then follow the disables that occur as a result to see what impact
this TPL disable has on the configuration of Futility.
CMake Error at cmake/tribits/core/package_arch/TribitsProcessEnabledTpl.cmake:127 (MESSAGE):
  ERROR: TPL_HDF5_NOT_FOUND=TRUE, aborting!
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake:1681 (TRIBITS_PROCESS_ENABLED_TPL)
  cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:202 (TRIBITS_PROCESS_ENABLED_TPLS)
  cmake/tribits/core/package_arch/TribitsProject.cmake:93 (TRIBITS_PROJECT_IMPL)
  cmake/Configure_Common.cmake:11 (TRIBITS_PROJECT)
  CMakeLists.txt:20 (INCLUDE)
-- Configuring incomplete, errors occurred!
See also "/build/CMakeFiles/CMakeOutput.log".
See also "/build/CMakeFiles/CMakeError.log".
make: *** No targets specified and no makefile found.  Stop.
Test project /build
No tests were found!!!
```

**Comment by bkochuna** [comment]
So, a couple things on this issue. Basically resolved it though.

The CI started failing when the TPLs were updated, but we did not update the `build_on_docker.sh` build script.

Travis-CI didn't register a failure because the CI script used `;` instead of `&&`. The former will chain bash commands and execute the chained commands regardless of the return code. The latter will only execute the command if the previous command's return code was 0. If it was non-zero, the non-zero return value is returned (in this case to Travis-CI).
